### PR TITLE
Anvil PM access

### PR DESCRIPTION
**Pull request by hanars**
Get PM user permissions from AnVIL instead of locally

### CODE REVIEW

File: seqr/views/apis/superuser_api_tests.py [hunk 1]
```diff
@@ -61,3 +71,13 @@ class AnvilSuperusersAPITest(AnvilAuthenticationTestCase, SuperusersAPITest):
...
 
+    def _test_pm_users(self, url):
+        # Test the case where the superuser does not have access to the PM group in AnVIL
+        # In that case, the request should succeed but not populate any PM users
+        response = self.client.get(url)
+        self._test_superuser_response(response, analyst_enabled=True, pm_enabled=False)
+
+        # Test if the superuser does have access to the PM group it populates properly
+        self.mock_get_group_members.side_effect = lambda *args, **kwargs: [self.pm_user.email]
+        super(AnvilSuperusersAPITest, self)._test_pm_users(url)
```
Comment (ShifaSZ) [inline seqr/views/apis/superuser_api_tests.py:82]: Add a test for `get_group_members` called?

File: seqr/views/apis/dashboard_api_tests.py [hunk 2]
```diff
@@ -102,6 +102,8 @@ def assert_has_anvil_calls(self):
     ]
     self.mock_get_ws_access_level.assert_has_calls(calls, any_order=True)
     self.mock_get_ws_acl.assert_not_called()
+    self.mock_get_groups.assert_not_called()
+    self.mock_get_group_members.assert_not_called()
```
Comment (ShifaSZ) [inline seqr/views/apis/dashboard_api_tests.py:106]: How about defining a function for these two assertions in the base class `AnvilAuthenticationTestCase`?

File: seqr/views/utils/pedigree_info_utils_tests.py [hunk 3]
```diff
@@ -118,7 +119,7 @@ def test_parse_sample_manifest(self, mock_email, mock_pm_group):
             '(i.e yes, no)', '(i.e yes, no, unknown)', 'ul', 'ng/ul', '', '', '', 'indicate study/protocol number']
 
         with self.assertRaises(ErrorsWarningsException) as ec:
-            parse_pedigree_table([header_1], FILENAME, user = User.objects.get(id=10))
+            parse_pedigree_table([header_1], FILENAME, user=self.analyst_user)
         self._assert_errors_warnings_exception(ec, f'Error while parsing file: {FILENAME}. Unsupported file format')
 
         user = User.objects.get(username='test_pm_user')
```
Comment (ShifaSZ) [inline seqr/views/utils/pedigree_info_utils_tests.py:125]: This line can be deleted.

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

File: seqr/views/utils/terra_api_utils.py [hunk 4]
```diff
@@ -128,7 +128,15 @@ def _get_service_account_access_token():
     return SERVICE_ACCOUNT_CREDENTIALS.token
 
 
-def anvil_call(method, path, access_token, user=None, headers=None, root_url=None, data=None, handle_errors=False):
+def anvil_call(method, path, access_token, user=None, headers=None, root_url=None, data=None, handle_errors=False,
+               cache_time=None, cache_key_id=None, process_response=None):
+    cache_key = f'terra_req__{cache_key_id or user}__{path}'
+    if cache_time:
```
[the remark this answers is not in the document]
Comment (hanars) [reply seqr/views/utils/terra_api_utils.py:134]: yes some functions cache for 60 seconds and some for 300

File: seqr/views/apis/project_api_tests.py [hunk 5]
```diff
@@ -502,10 +502,15 @@ class AnvilProjectAPITest(AnvilAuthenticationTestCase, ProjectAPITest):
     PROJECT_COLLABORATORS = ANVIL_COLLABORATORS
     HAS_EMPTY_PROJECT = False
 
+
```
[the remark this answers is not in the document]
Comment (hanars) [reply seqr/views/apis/project_api_tests.py:505]: yes needed, by PEP standards there should have been 2 blank lines here but was only 1

**Review by ShifaSZ — APPROVED** [review]

**Pull request merged**